features/step_definitions/barclamp_pacemaker/pacemaker_steps.rb

```diff
@@ -0,0 +1,292 @@
+    puts "Available Stonith Devices: #{response}"
+    @stonith_device = response.split(/\n/)[0].strip!
+    puts "Obtained Stonith Device: #{@stonith_device}"
+    expect(@stonith_device).to eq("stonith-#{@secondary_node.attributes()[:name]}")
+end
+
+And(/^I can obtain metadata for an arbitrary fencing agent$/) do
+    response = @primary_node.exec!("stonith_admin -M #{@stonith_device} --agent fence_pcmk").output
+    puts "Stonith device Metadata: #{response}"
+    expect(response).not_to be_empty
+end
+
+And(/^I can check if cluster is configured with stonith-enabled$/) do
+    response = @primary_node.exec!("crm configure show type:property").output
+    puts "CRM configuration: #{response}"
+    expect(response).not_to be_empty
+end
+
+And(/^I can check the failcount of stonith resources$/) do
+    response = @primary_node.exec!("crm_failcount --resource-id=rabbitmq").output
+    puts "CRM Failcount: #{response}"
+    fail_count = response.tr("\n", "").split(/ /)[-1]
+    puts "Fail count : #{fail_count}"
+    expect(fail_count).to eq("value=0")
+end
+
+And(/^I can ensure that start operations for haproxy, rabbitmq, postgresql are complete$/) do
+    node_to_check = @primary_node.attributes()[:name]
+    resource_list = ["rabbitmq", "haproxy", "postgresql"]
+    resource_list.each do |resource_name|
+        response = @primary_node.exec!("crm_resource --list-operations --resource #{resource_name} --node #{node_to_check}").output
+        puts "Resource response: #{response}"
+        status = response.tr("\n","").split(/ /)[-1]
+        expect(status).to eq("complete")
+    end
+end
+
+And(/^I can update and delete new attribute to a crm node$/) do
+    @primary_node.exec!("crm_attribute --node d52-54-01-77-77-01 --name location --update office")
+    @primary_node.exec!("crm_attribute --node d52-54-01-77-77-01 --name location --query")
+    @primary_node.exec!("crm_attribute --node d52-54-01-77-77-01 --name location --delete")
+end
+
+And(/^I can verify the status of corosync being active with no faults and correct ring addresses$/) do
+    prim_ring_addr = ""
+    prim_status = ""
+    sec_ring_addr = ""
+    sec_status = ""
+   
+    # Get ring address from node configuration
+    response = @primary_node.exec!("crm corosync get nodelist.node.ring0_addr").output
+    ring_addr_conf = response.split(/\n/)
+    puts "Ring Addresses from configure: #{ring_addr_conf}"
+
+    # Get ring status from primary node
+    response = @primary_node.exec!("crm corosync status").output
+    puts "Primary Node Corosync: #{response}"
+    lines = response.tr("\t", " ").split(/\n/)
+    lines.each do |line|
+        if line.include?("id =")
+            prim_ring_addr = line.split(/=/)[-1].strip!
+        end
+        if line.include?("status =")
+            prim_status = line.split(/=/)[-1].strip!
+        end
+    end
+
+    # Get ring status from secondary node
+    response = @secondary_node.exec!("crm corosync status").output
```

Comment:
This chunk of code is almost identical to the one above - please can you extract into a separate method and reuse it?